src/DotNetLightning.Core/Channel/ChannelHelpers.fs

```diff
@@ -17,3 +24,439 @@ module internal ChannelConstantHelpers =
+
+    module RevokedClose =
+        let private ClaimMainOutput
+            (closingTx: Transaction)
+            (commitmentNumber: CommitmentNumber)
+            (staticChannelConfig: StaticChannelConfig)
+            (remotePerCommitmentSecret: Choice<PerCommitmentSecret, PerCommitmentSecretStore>)
+            (localChannelPrivKeys: ChannelPrivKeys): Result<TransactionBuilder, OutputClaimError>
+            =
+            result {
+                let localChannelPubKeys = localChannelPrivKeys.ToChannelPubKeys()
+                let remoteChannelPubKeys = staticChannelConfig.RemoteChannelPubKeys
+
+                let! perCommitmentSecret =
+                    match remotePerCommitmentSecret with
+                    | Choice1Of2 remotePerCommitmentSecret -> Ok remotePerCommitmentSecret
+                    | Choice2Of2 remotePerCommitmentSecretStore ->
+                        let commitmentSecretOpt =
+                            remotePerCommitmentSecretStore.GetPerCommitmentSecret commitmentNumber
+
+                        match commitmentSecretOpt with
+                        | Some commitmentSecret -> Ok commitmentSecret
+                        | None -> Error OutputClaimError.UnknownClosingTx
+
+                let perCommitmentPoint = perCommitmentSecret.PerCommitmentPoint()
+
+                let localCommitmentPubKeys =
+                    perCommitmentPoint.DeriveCommitmentPubKeys localChannelPubKeys
+
+                let remoteCommitmentPubKeys =
+                    perCommitmentPoint.DeriveCommitmentPubKeys remoteChannelPubKeys
+
+                let transactionBuilder =
+                    staticChannelConfig.Network.CreateTransactionBuilder()
+
+                let toRemoteScriptPubKey =
+                    localCommitmentPubKeys
+                        .PaymentPubKey
+                        .RawPubKey()
+                        .WitHash
+                        .ScriptPubKey
+
+                let toLocalScriptPubKey =
+                    Scripts.toLocalDelayed
+                        localCommitmentPubKeys.RevocationPubKey
+                        staticChannelConfig.LocalParams.ToSelfDelay
+                        remoteCommitmentPubKeys.DelayedPaymentPubKey
+
+                let toLocalWitScriptPubKey = toLocalScriptPubKey.WitHash.ScriptPubKey
+
+                let toRemoteIndexOpt =
+                    closingTx.Outputs
+                    |> Seq.tryFindIndex (fun out -> out.ScriptPubKey = toRemoteScriptPubKey)
+
+                toRemoteIndexOpt
+                |> Option.iter
+                    (
+                        fun toRemoteIndex ->
+                            let localPaymentPrivKey =
+                                perCommitmentPoint.DerivePaymentPrivKey
+                                    localChannelPrivKeys.PaymentBasepointSecret
+
+                            transactionBuilder.SetVersion TxVersionNumberOfCommitmentTxs
+                            |> ignore<TransactionBuilder>
```

Comment:
nit: why not just `ignore` ?

Reply:
We have a policy wrt this, the reason (short answer) I think is to not accidentally ignore a function (e.g. int -> TransactionBuilder) instead of the outcome/result of the function. @knocte can give you a more complete answer.
Ah also there's a tweet about this: https://twitter.com/ChetHusk/status/1351156012590526474

Reply:
Ah I see, I think I will follow this rule too from now on.